ibc/types.go

```diff
@@ -35,6 +35,10 @@ type ChainConfig struct {
 	TrustingPeriod string `yaml:"trusting-period"`
 	// Do not use docker host mount.
 	NoHostMount bool `yaml:"no-host-mount"`
+	// When true, will skip validator gentx flow
+	SkipGenTx *bool
```

Comment:
What's the reasoning for making this a pointer? Looks like we are just checking if this is false and if so we run the usual gentx logic. Couldn't we just use the fact that the zero value is false and gentx will only be skipped if explicitly set to true?

Reply:
The reason for the pointer is so that during `MergeChainSpecConfig` and `applyConfigOverrides`, a zero value `false` would not override a prior intentional `true`. Having `nil` as the zero value will confirm override intent when the user has explicitly set the value to a non-nil bool.

With a non-pointer, it would probably work well enough in those mentioned methods to only use the incoming value if it's `true`, but this would mean it would not be possible to override a `true` value with `false` (any `true` value throughout means it's `true`). This is good enough for the current use case of this though, and does simplify the usage when declaring `ChainConfig` instances.

Reply:
thanks for the reasoning! makes sense to me.